conn.go

```diff
@@ -658,17 +671,26 @@ func (c *Conn) ReadMessage(maxBytes int) (Message, error) {
 // gives the minimum and maximum number of bytes that it wants to receive from
 // the kafka server.
 func (c *Conn) ReadBatch(minBytes, maxBytes int) *Batch {
+	cfg := BatchReadConfig{
+		MinBytes: minBytes,
+		MaxBytes: maxBytes,
+	}
+	return c.ReadBatchWithConfig(cfg)
+}
+
+func (c *Conn) ReadBatchWithConfig(cfg BatchReadConfig) *Batch {
```

Comment:
```suggestion
func (c *Conn) ReadBatchWith(cfg ReadBatchConfig) *Batch {
```
- We have `NewConnWith` already, so let's match that naming scheme and call the method `ReadBatchWith`?
- To follow these naming schemes, how about also calling the configuration type `ReadBatchConfig`?